src/register/containers/register/register.component.ts

```diff
@@ -62,15 +72,15 @@ export class RegisterComponent implements OnInit, OnDestroy {
     } else {
       this.isPageValid = false;
       const { value } = formDraft;
-      const nextUrl = value.nextUrl;
-      delete value.nextUrl; // removing nextUrl so ti doesn't overwrite the one from the server payload.
-      this.store.dispatch(new fromStore.SaveFormData({value, nextUrl}));
+      this.store.dispatch(new fromStore.SaveFormData({value, pageId: this.pageId}));
+
     }
   }
 
   ngOnDestroy(): void {
     this.$pageItemsSubscription.unsubscribe();
     this.$routeSubscription.unsubscribe();
+    this.$nextUrlSubscription.unsubscribe();
```

Comment:
please check if the object is not null/undefined before unsubscribe.

Reply:
there is no need to do that... this a subscription it is never null 
state.pipe(select is always an object)

Reply:
Okay, nevertheless its a good practice.